ext/node/polyfills/http.ts

```diff
@@ -455,25 +450,30 @@ class ClientRequest extends OutgoingMessage {
       this._bodyWriteRid = resourceForReadableStream(readable);
     }
 
-    this._req = op_node_http_request(
-      this.method,
-      url,
-      headers,
-      client[internalRidSymbol],
-      this._bodyWriteRid,
-    );
-
     (async () => {
       try {
-        const res = await op_node_http_fetch_send(this._req.requestRid);
-        if (this._req.cancelHandleRid !== null) {
-          core.tryClose(this._req.cancelHandleRid);
-        }
-        if (this._timeout) {
-          this._timeout.removeEventListener("abort", this._timeoutCb);
-          webClearTimeout(this._timeout[timerId]);
+        const parsedUrl = new URL(url);
+        let baseConnRid =
+          this.socket._handle[kStreamBaseField][internalRidSymbol];
+        if (this._encrypted) {
+          [baseConnRid] = op_tls_start({
+            rid: baseConnRid,
+            hostname: parsedUrl.hostname,
+            caCerts: [],
+            alpnProtocols: ["http/1.0", "http/1.1"],
+          });
         }
-        this._client.close();
+        const rid = await op_node_http_request_with_conn(
+          this.method,
+          url,
+          headers,
+          this._bodyWriteRid,
+          baseConnRid,
+          this._encrypted,
+        );
+        // Emit request ready to let the request body to be written.
+        this.emit("requestReady");
```

Comment:
Is this an internal event? I can't seem to find it in https://nodejs.org/api/http.html or Node codebase. If so, can you think of other mechanism we could do it (a deferred maybe?) so that it's not visible to users?

Reply:
Probably this is only necessary for calling `_flushBuffer()`. I'll try that idea